### PR TITLE
[GNA] Restore deprecated options for GNA Plugin

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Restore deprecatet options for GNA Plugin

### Tickets:
 - 75690

### CODE REVIEW

File: src/inference/include/ie/gna/gna_config.hpp [hunk 1]
```diff
@@ -65,6 +70,19 @@ DECLARE_GNA_CONFIG_VALUE(SW);
...
+ * @brief Modes GNA_GEN, GNA_GEN_EXACT GNA_SSE, GNA_SSE_EXACT, GNA_AVX1, GNA_AVX1_EXACT, GNA_AVX2, GNA_AVX2_EXACT are deprecated
+ * and will be removed in 2022.2
+ */
+DECLARE_GNA_CONFIG_VALUE(GEN);
+DECLARE_GNA_CONFIG_VALUE(GEN_EXACT);
+DECLARE_GNA_CONFIG_VALUE(SSE);
+DECLARE_GNA_CONFIG_VALUE(SSE_EXACT);
+DECLARE_GNA_CONFIG_VALUE(AVX1);
+DECLARE_GNA_CONFIG_VALUE(AVX1_EXACT);
+DECLARE_GNA_CONFIG_VALUE(AVX2);
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/gna/gna_config.hpp:83]: please, deprecated using `INFERENCE_ENGINE_DEPRECATED` macro

Comment (kbruniec) [reply src/inference/include/ie/gna/gna_config.hpp:91]: done

**Comment by dorloff** [comment]
@kbruniec, we discussed with @nkogteva that it makes sense to deprecate the options PWL_UNIFORM_DESIGN and PWL_MAX_ERROR_PERCENT, too, because we are planning to get rid of them as well. Could you please do this?

**Comment by kbruniec** [comment]
> @kbruniec, we discussed with @nkogteva that it makes sense to deprecate the options PWL_UNIFORM_DESIGN and PWL_MAX_ERROR_PERCENT, too, because we are planning to get rid of them as well. Could you please do this?

@dorloff Done

File: src/inference/include/ie/gna/gna_config.hpp [hunk 2]
```diff
@@ -96,13 +124,15 @@ DECLARE_GNA_CONFIG_KEY(COMPACT_MODE);
...
  */
+INFERENCE_ENGINE_DEPRECATED("The config key is deprected and will be removed")
 DECLARE_GNA_CONFIG_KEY(PWL_UNIFORM_DESIGN);
 
 /**
  * @brief The option to allow to specify the maximum error percent that the optimized algorithm finding
  * will use to find PWL functions.
  * By default (in case of NO value set), 1.0 value is used.
  */
+INFERENCE_ENGINE_DEPRECATED("The config key is deprected and will be removed")
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/gna/gna_config.hpp:135]: add doxygen deprecation message

File: src/inference/include/ie/gna/gna_config.hpp [hunk 3]
```diff
@@ -96,13 +124,15 @@ DECLARE_GNA_CONFIG_KEY(COMPACT_MODE);
  * If value is YES then simple uniform distribution used to create PWL approximation of activation functions
  * Uniform distribution usually gives poor approximation with same number of segments
  */
+INFERENCE_ENGINE_DEPRECATED("The config key is deprected and will be removed")
```
Comment (ilya-lavrenov) [inline src/inference/include/ie/gna/gna_config.hpp:127]: add doxygen as well

File: src/inference/include/ie/gna/gna_config.hpp (hunk 2 above)
Comment (kbruniec) [reply src/inference/include/ie/gna/gna_config.hpp:180]: done

File: src/inference/include/ie/gna/gna_config.hpp (hunk 3 above)
Comment (kbruniec) [reply src/inference/include/ie/gna/gna_config.hpp:171]: done

File: src/inference/include/ie/gna/gna_config.hpp [hunk 4]
```diff
@@ -95,14 +166,18 @@ DECLARE_GNA_CONFIG_KEY(COMPACT_MODE);
...
+ * @deprecated The config key is deprecated and will be removed in 2022.2.
  */
+INFERENCE_ENGINE_DEPRECATED("The config key is deprected and will be removed")
 DECLARE_GNA_CONFIG_KEY(PWL_UNIFORM_DESIGN);
 
 /**
  * @brief The option to allow to specify the maximum error percent that the optimized algorithm finding
  * will use to find PWL functions.
  * By default (in case of NO value set), 1.0 value is used.
+ * @deprecated The config key is deprecated and will be removed in 2022.2.
```
Comment (nkogteva) [inline src/inference/include/ie/gna/gna_config.hpp:178]:
```suggestion
 * @deprecated The config key is deprecated and will be removed in a future release
```

File: src/inference/include/ie/gna/gna_config.hpp [hunk 5]
```diff
@@ -53,6 +53,13 @@ DECLARE_GNA_CONFIG_KEY(PRECISION);
  */
 DECLARE_GNA_CONFIG_KEY(FIRMWARE_MODEL_IMAGE);
 
+/**
+ * @brief Generation of GNA embedded device to export the model.
+ * @deprecated Key is deprecated and will be removed in 2022.2.
```
Comment (dorloff) [inline src/inference/include/ie/gna/gna_config.hpp:58]: Let's put it this way: "will be removed in a future release". I would not commit that each option will be removed in 2022.2.

File: src/inference/include/ie/gna/gna_config.hpp (hunk 4 above)
Comment (kbruniec) [reply src/inference/include/ie/gna/gna_config.hpp]: done

File: src/inference/include/ie/gna/gna_config.hpp (hunk 5 above)
Comment (kbruniec) [reply src/inference/include/ie/gna/gna_config.hpp]: done

File: src/plugins/intel_gna/gna_plugin_config.cpp [hunk 6]
```diff
@@ -121,7 +134,9 @@ void Config::UpdateFromMap(const std::map<std::string, std::string>& config) {
                                     << value;
             }
             gnaPrecision = precision;
+            IE_SUPPRESS_DEPRECATED_START
         } else if (key == GNA_CONFIG_KEY(PWL_UNIFORM_DESIGN)) {
+            // This key is deprecated and will be removed in 2022.1
```
Comment (dorloff) [inline src/plugins/intel_gna/gna_plugin_config.cpp:139]: Still 2022.1.

File: src/plugins/intel_gna/gna_plugin_config.cpp [hunk 7]
```diff
@@ -133,6 +148,8 @@ void Config::UpdateFromMap(const std::map<std::string, std::string>& config) {
                                     << "should be equal to YES/NO, but not" << value;
             }
         } else if (key == GNA_CONFIG_KEY(PWL_MAX_ERROR_PERCENT)) {
+            // This key is deprecated and will be removed in 2022.1
```
Comment (dorloff) [inline src/plugins/intel_gna/gna_plugin_config.cpp:151]: Still 2022.1.

Comment (kbruniec) [reply src/plugins/intel_gna/gna_plugin_config.cpp]: Right, fixed

File: src/plugins/intel_gna/gna_plugin_config.cpp (hunk 6 above)
Comment (kbruniec) [reply src/plugins/intel_gna/gna_plugin_config.cpp]: Right, fixed

**Review by nkogteva — APPROVED** [review]

**Pull request merged**